projects/igniteui-angular/src/lib/data-operations/filtering-util.ts

```diff
@@ -0,0 +1,127 @@
+import { IgxBooleanFilteringOperand, IgxDateFilteringOperand, IgxDateTimeFilteringOperand, IgxFilteringOperand, IgxNumberFilteringOperand, IgxStringFilteringOperand, IgxTimeFilteringOperand } from './filtering-condition';
+import { IFilteringExpression, ISerializedFilteringExpression } from './filtering-expression.interface';
+import { FilteringExpressionsTree, IFilteringExpressionsTree, ISerializedFilteringExpressionTree } from './filtering-expressions-tree';
+
+export class FilteringUtil {
+    /**
+     * Recreates a `FilteringExpressionsTree` from a given JSON serialized string.
+     * @param jsonString The serialized string to recreate the tree from.
+     * @returns A fully reconstructed `FilteringExpressionsTree`.
+     */
+    public static recreateTreeFromJson(jsonString: string): FilteringExpressionsTree {
```

Comment:
Attempt to do:
var tree1 = FilteringUtil.recreateTreeFromJson(JSON.stringify(this.expressionTree));

This method throws the following error:
ERROR TypeError: Cannot read properties of undefined (reading 'toString')
    at filtering-expressions-tree.ts:230:55
    at Array.find (<anonymous>)
    at containsCondition (filtering-expressions-tree.ts:229:40)
    at filtering-expressions-tree.ts:244:43
    at Array.forEach (<anonymous>)
    at filtering-expressions-tree.ts:243:42
    at Array.forEach (<anonymous>)
    at packTree (filtering-expressions-tree.ts:237:37)
    at filtering-expressions-tree.ts:252:49
    at Array.forEach (<anonymous>)

Reply:
I've added a check to prevent this.